lib/src/notifiers/hn_api.dart

```diff
@@ -71,13 +81,20 @@ class HackerNewsTab with ChangeNotifier {
     notifyListeners();
     loadingTabsCount.value += 1;
 
-    final worker = Worker();
-    await worker.isReady;
+    try {
+      final refreshedArticles = await executorService.submit(
+        FetchStories(storiesType, cachedArticles: _cachedArticles),
+      );
+
+      for (final article in refreshedArticles) {
+        _cachedArticles[article.id] = article;
+      }
 
-    _articles = await worker.fetch(storiesType);
-    _isLoading = false;
+      _articles = refreshedArticles;
+    } on HackerNewsApiException catch (error) {
```

Comment:
We were not handling error cases so i added a catch clause but it's like CI does not like the unused error variable.